performance_test/script.mjs

```diff
@@ -0,0 +1,98 @@
+import { execSync } from "child_process";
+import { copyFile } from "fs/promises";
+import { performance } from "perf_hooks";
+import { existsSync, mkdirSync, rmdirSync } from "fs";
+
+const DEFAULT_NUMBER_OF_FILE = 5;
+const DEFAULT_ITERATION = 1;
+const BASE_PATH = "./performance_test";
+const DEFAULT_MOCK_FILE_PATH = `${BASE_PATH}/mock_unwrapped_doc.json`;
+const INPUT_MOCK_UNWRAPPED_FILE_FOLDER = `${BASE_PATH}/raw-documents`;
+const OUTPUT_MOCK_WRAPPED_FILE_FOLDER = `${BASE_PATH}/wrapped-documents`;
+
+// Setup number of files using mocked json
+const setup = async (file_path, numberOfFiles) => {
+  var fileNameArray = file_path.split("/").pop();
+  var mockFileName = fileNameArray.split(".")[0];
+  var mockFileExtension = fileNameArray.split(".")[1];
+  try {
+    existsSync(INPUT_MOCK_UNWRAPPED_FILE_FOLDER) || mkdirSync(INPUT_MOCK_UNWRAPPED_FILE_FOLDER);
+    for (var index = 0; index < numberOfFiles; index++) {
+      await copyFile(
+        file_path,
+        `${INPUT_MOCK_UNWRAPPED_FILE_FOLDER}/${mockFileName + (index + 1)}.${mockFileExtension}`
+      );
+    }
+  } catch (e) {
+    console.error(e);
+  }
+};
+
+// Destroy generated folder
+const destroy = () => {
+  !existsSync(INPUT_MOCK_UNWRAPPED_FILE_FOLDER) || rmdirSync(INPUT_MOCK_UNWRAPPED_FILE_FOLDER, { recursive: true });
+  !existsSync(OUTPUT_MOCK_WRAPPED_FILE_FOLDER) || rmdirSync(OUTPUT_MOCK_WRAPPED_FILE_FOLDER, { recursive: true });
+};
+
+// Monitoring wrap function
+const monitor = () => {
+  return new Promise((resolve, reject) => {
+    try {
+      // Start monitoring
+      var startTime = performance.now();
+
+      // Call OA CLI wrap
+      execSync(
+        `open-attestation wrap ${INPUT_MOCK_UNWRAPPED_FILE_FOLDER} --output-dir ${OUTPUT_MOCK_WRAPPED_FILE_FOLDER}`,
```

Comment:
this means the user MUST have the cli installed and available in its path. Let's switch to typescript, so that we can import the function performing wrapping instead.

Also it will be a problem in the ci :)

Reply:
done. i have changed to call the wrap implementation directly, however the response timing will be affected slightly as it skipped some of the initial checks happening at commands input.